File: KendoExperiments/Scripts/review/models/question.model.ts
module Experiments.Models {
 
    export class QuestionModel extends kendo.data.ObservableObject {
        id: number;
        name: string;
        questionType: string;
        availableList: string;
        answer: string;

        constructor(value?: any) {
            super(value);
        }

        static createInstance(data: any): QuestionModel {
            if (data.questionType == "T")
                return new QuestionAnswerTextModel(data);

            return new QuestionModel(data);
        }
    }
    
    export class QuestionAnswerModel extends QuestionModel {
        answer: any;

        constructor(value?: any) {
            super(value);
        }
    }
    
    export class QuestionAnswerTextModel extends QuestionAnswerModel {
        answer: string;

        constructor(value?: any) {
            super(value);
        }
    }
    
    export class QuestionAnswerListModel extends QuestionAnswerModel {
        answer: Array<any>;

        constructor(value?: any) {
            super(value);

        }
    }   
} 

File: KendoExperiments/api/Models/QuestionModel.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace KendoExperiments.api.Models
{
    public class QuestionModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Order { get; set; }
        public string Answer { get; set; }
        public string QuestionType { get; set; }
        public string AvailableList { get; set; }
        public DateTime ModifiedDate { get; set; }
    }
}